### PR TITLE
feat: Search functionality

### PR DESCRIPTION
**Pull request by PeeyushPrashant**
Screenshot (99)
Screenshot (100)

### CODE REVIEW

**Review by vinc3nati — APPROVED** [review]
Great feature, code looks concise.

File: src/components/NavBar/NavBar.jsx [hunk 1]
```diff
@@ -16,7 +18,16 @@ export const NavBar=()=>{
             />
             <small className="nav-heading-small">Note</small>
           </div>
-         
+          <div className="nav-search flex-row">
+          <i className="fas fa-search search-icon"></i>
+          <input type="text" className="nav-input" placeholder="Type to search"
+          onKeyDown={(e)=>{
+            if(e.key === 'Enter' || e.target.value === ''){
```
Comment (ramalingaa) [inline src/components/NavBar/NavBar.jsx:25]: Good implemention on backspace, user doesn't have to search again

**Review by ramalingaa — APPROVED** [review]
Clean code and nicely implemented logic.This PR can be merged

File: src/context/data-context.js [hunk 2]
```diff
@@ -32,18 +45,27 @@ const DataProvider = ({ children }) => {
...
+  };
   const [noteState, noteDispatch] = useReducer(noteReducer, { noteData: [] });
   const [archiveState, archiveDispatch] = useReducer(archiveReducer, {
     archiveData: [],
   });
   const [note, setNote] = useState(initialVal);
+  const [filterState, filterDispatch] = useReducer(filterReducer, {
+    filter: initialFilter,
+  });
+  // console.log(filterState.filter);
```
Comment (Rohanmond) [inline src/context/data-context.js:59]: remove comments

**Review by Rohanmond — APPROVED** [review]
code is looking good to merge

**Pull request merged**